ddtrace/contrib/internal/langchain/utils.py

```diff
@@ -0,0 +1,79 @@
+import inspect
+import sys
+
+
+class BaseTracedLangChainStreamResponse:
+    def __init__(self, generator, integration, span, on_span_finish):
+        self._generator = generator
+        self._dd_integration = integration
+        self._dd_span = span
+        self._on_span_finish = on_span_finish
+        self._chunks = []
```

Comment:
Looking at langchain's [source code](https://github.com/langchain-ai/langchain/blob/63c3cc1f1f2b78eb588e56fb62a7dacb43a2e27f/libs/core/langchain_core/language_models/chat_models.py#L408-L422), it looks like they actually add the chunks for us in the stream() operation. Would it be possible for us to access/reuse that instead of performing the appending/processing ourselves?

Reply:
good find! that one looks specific to chat models, i'll check if their runnable sequence (chain) implements something similar.

Reply:
it looks like it would only be accessible through the `run_manager` instance, which relies on the callback configs. i think we've thought about patching callback configs to run our own callbacks (which would good for handling these kinds of events for span management), but i think it's a bit out of scope for this PR. otherwise, it doesn't look like that accumulated `generations` goes anywhere we'd be able to use it 😢